File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/dtos/ChangePasswordDto.java
package com.example.dalathasfarm.dtos;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChangePasswordDto {
    @NotBlank(message = "Mật khẩu cũ là bắt buộc")
    private String oldPassword;

    @NotBlank(message = "Mật khẩu mới là bắt buộc")
    private String newPassword;

    @NotBlank(message = "Nhập lại mật khẩu mới là bắt buộc")
    private String retypeNewPassword;
}

File: frontend/src/app/dtos/user/change-password.dto.ts
export class ChangePasswordDto {
    oldPassword: string;
    newPassword: string;
    retypeNewPassword: string;

    constructor(data: any) {
        this.oldPassword = data.oldPassword;
        this.newPassword = data.newPassword;
        this.retypeNewPassword = data.retypeNewPassword;
    }
}